### PR TITLE
Added user(identity) existence validation for manager/tenant registration

### PR DESCRIPTION
**Pull request by imilovanovic**
Added user(identity) existence validation for manager/tenant registration.

### CODE REVIEW

File: query/src/main/java/eu/execom/buildingmanager/query/authentication/CredentialTransformer.java [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
   @Override
   public Credential from(Authentication document) {
+    if(null == document) {
```
Comment (imilovanovic) [inline query/src/main/java/eu/execom/buildingmanager/query/authentication/CredentialTransformer.java]: OK, to fix

File: query/src/main/java/eu/execom/buildingmanager/query/validation/IdentityUniqueValidatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    CompletableFuture<Credential> result = queryGateway.send(query, Credential.class);
+    Credential credentials = null;
+    try {
+      credentials = result.get();
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Failed to check identity uniqueness (interrupted error).",e);
+    } catch (ExecutionException e) {
+      throw new RuntimeException("Failed to check identity uniqueness (execution error).",e);
+    }
+    return null == credentials;
```
Comment (imilovanovic) [inline query/src/main/java/eu/execom/buildingmanager/query/validation/IdentityUniqueValidatorImpl.java]: to fix

File: domain/src/main/java/eu/execom/buildingmanager/domain/credentials/IdentityUniqueValidator.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package eu.execom.buildingmanager.domain.credentials;
+
+import eu.execom.buildingmanager.domain.utility.Validator;
+
+@FunctionalInterface
+public interface IdentityUniqueValidator extends Validator<String> {
```
Comment (stanislavgrujic) [inline domain/src/main/java/eu/execom/buildingmanager/domain/credentials/IdentityUniqueValidator.java]: Why not name it `UniqueIdentityValidator`, it sounds more expressive.

File: query/src/main/java/eu/execom/buildingmanager/query/authentication/CredentialTransformer.java [hunk 4]
```diff
@@ -9,6 +9,9 @@
 
   @Override
   public Credential from(Authentication document) {
+    if(document == null) {
```
Comment (stanislavgrujic) [inline query/src/main/java/eu/execom/buildingmanager/query/authentication/CredentialTransformer.java]: Please configure your formatter in IDE to properly format the code. You can find the configuration file in `conf/formatter.xml`.

File: query/src/main/java/eu/execom/buildingmanager/query/validation/IdentityUniqueValidatorImpl.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  public boolean isValid(String email) {
+    GetCredentialsByEmailQuery query = new GetCredentialsByEmailQuery(email);
+    CompletableFuture<Credential> result = queryGateway.send(query, Credential.class);
+    Credential credentials = null;
+    try {
+      credentials = result.get();
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Failed to check identity uniqueness (interrupted error).",e);
+    } catch (ExecutionException e) {
+      throw new RuntimeException("Failed to check identity uniqueness (execution error).",e);
```
Comment (stanislavgrujic) [inline query/src/main/java/eu/execom/buildingmanager/query/validation/IdentityUniqueValidatorImpl.java]: Please fix formatting.

File: domain/src/main/java/eu/execom/buildingmanager/domain/credentials/IdentityUniqueValidator.java (hunk 3 above)
Comment (imilovanovic) [reply domain/src/main/java/eu/execom/buildingmanager/domain/credentials/IdentityUniqueValidator.java]: Depends on question in "developer's head": "Is your identity unique?" or "Do you have unique identity?" :)

**Pull request merged**